pykeops/torch/utils.py

```diff
@@ -157,6 +172,122 @@ def device(x):
         else:
             return None
 
+    @staticmethod
+    def distance_function(metric):
+        def euclidean(x, y):
+            return ((x - y) ** 2).sum(-1)
+
+        def manhattan(x, y):
+            return ((x - y).abs()).sum(-1)
+
+        def angular(x, y):
+            return -(x | y)
+
+        def angular_full(x, y):
+            return angular(x, y) / ((angular(x, x) * angular(y, y)).sqrt())
```

Comment:
The comments for the NumPy version also apply to this file.

Reply:
done